## Estadística Aplicada

Actividad 6 parte A

Alexandra Guadalupe Valdez Mireles

1910361       

Gpo 44

In [101]:
## Librerías Básicas
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
%matplotlib inline

In [102]:
## Librerías Especializadas
from lifelines import CoxPHFitter, KaplanMeierFitter, ExponentialFitter, WeibullFitter, LogNormalFitter
import seaborn as sns

In [103]:
## Importar Datos 
df_desercion = pd.read_csv('https://raw.githubusercontent.com/jimmyzac/Estadistica-Aplicada-FCFM-UANL/main/bases_datos/desercion_laboral.csv')

In [104]:
df_desercion.head()

satisfaccion  evaluacion  proyectos  promedio_horas_mes  semestre  \
0           3.8         5.3          2                 157         3   
1           8.0         8.6          5                 262         6   
2           1.1         8.8          7                 272         4   
3           7.2         8.7          5                 223         5   
4           3.7         5.2          2                 159         3   

   accidente  renunciar  ascenso_5years departamento salario  
0          0          1               0       ventas    bajo  
1          0          1               0       ventas   medio  
2          0          1               0       ventas   medio  
3          0          1               0       ventas    bajo  
4          0          1               0       ventas    bajo

### DEPURAR LA BASE DE DATOS Y ESTADISTICAS DESCRIPTIVAS

In [105]:
## Verificar missing values (valores perdidos)
df_desercion.isnull().sum()

satisfaccion           0
evaluacion             0
proyectos              0
promedio_horas_mes     0
semestre               0
accidente              0
renunciar              0
ascenso_5years         0
departamento          26
salario                0
dtype: int64

In [106]:
## Verificar que no haya valores duplicados
df_desercion.duplicated().sum()

2982

In [107]:
## Eliminar valores duplicados
df_desercion=df_desercion.drop_duplicates(keep='first').reset_index(drop=True)

In [108]:
df_desercion.head()

satisfaccion  evaluacion  proyectos  promedio_horas_mes  semestre  \
0           3.8         5.3          2                 157         3   
1           8.0         8.6          5                 262         6   
2           1.1         8.8          7                 272         4   
3           7.2         8.7          5                 223         5   
4           3.7         5.2          2                 159         3   

   accidente  renunciar  ascenso_5years departamento salario  
0          0          1               0       ventas    bajo  
1          0          1               0       ventas   medio  
2          0          1               0       ventas   medio  
3          0          1               0       ventas    bajo  
4          0          1               0       ventas    bajo

In [109]:
## Verificar nuevamente que no haya valores duplicados
df_desercion.duplicated().sum()

0

In [110]:
## Verificar que las variables sean numéricas
df_desercion.dtypes

satisfaccion          float64
evaluacion            float64
proyectos               int64
promedio_horas_mes      int64
semestre                int64
accidente               int64
renunciar               int64
ascenso_5years          int64
departamento           object
salario                object
dtype: object

In [111]:
## Convertir a dummies
dummies=pd.get_dummies(df_desercion['departamento'], prefix='depa').astype(int)
dummies.head()

depa_Investigación  depa_administración  depa_contabilidad  \
0                   0                    0                  0   
1                   0                    0                  0   
2                   0                    0                  0   
3                   0                    0                  0   
4                   0                    0                  0   

   depa_desarrollo de producto  depa_informática  depa_mantenimiento  \
0                            0                 0                   0   
1                            0                 0                   0   
2                            0                 0                   0   
3                            0                 0                   0   
4                            0                 0                   0   

   depa_mercadotecnia  depa_recursos humanos  depa_soporte  depa_ventas  
0                   0                      0             0            1  
1                   0                      0             0            1  
2                   0                      0             0            1  
3                   0                      0             0            1  
4                   0                      0             0            1

In [112]:
dummies.describe()

depa_Investigación  depa_administración  depa_contabilidad  \
count        12017.000000         12017.000000       12017.000000   
mean             0.057752             0.036282           0.051677   
std              0.233283             0.186999           0.221383   
min              0.000000             0.000000           0.000000   
25%              0.000000             0.000000           0.000000   
50%              0.000000             0.000000           0.000000   
75%              0.000000             0.000000           0.000000   
max              1.000000             1.000000           1.000000   

       depa_desarrollo de producto  depa_informática  depa_mantenimiento  \
count                 12017.000000      12017.000000        12017.000000   
mean                      0.057086          0.081218            0.186735   
std                       0.232016          0.273181            0.389715   
min                       0.000000          0.000000            0.000000   
25%                       0.000000          0.000000            0.000000   
50%                       0.000000          0.000000            0.000000   
75%                       0.000000          0.000000            0.000000   
max                       1.000000          1.000000            1.000000   

       depa_mercadotecnia  depa_recursos humanos  depa_soporte   depa_ventas  
count        12017.000000           12017.000000  12017.000000  12017.000000  
mean             0.056004               0.050012      0.151535      0.269535  
std              0.229939               0.217980      0.358585      0.443737  
min              0.000000               0.000000      0.000000      0.000000  
25%              0.000000               0.000000      0.000000      0.000000  
50%              0.000000               0.000000      0.000000      0.000000  
75%              0.000000               0.000000      0.000000      1.000000  
max              1.000000               1.000000      1.000000      1.000000

In [113]:
#Se guardan los dummies en el dataframe y se eliminan los datos reales de departamento
dummies = dummies[['depa_Investigación','depa_administración','depa_contabilidad', 'depa_desarrollo de producto', 'depa_informática', 'depa_mantenimiento', 'depa_mercadotecnia', 'depa_recursos humanos', 'depa_soporte']]
df_desercion = pd.concat([df_desercion,dummies],axis=1)
df_desercion = df_desercion.drop('departamento',axis=1)

In [114]:
df_desercion.head()

satisfaccion  evaluacion  proyectos  promedio_horas_mes  semestre  \
0           3.8         5.3          2                 157         3   
1           8.0         8.6          5                 262         6   
2           1.1         8.8          7                 272         4   
3           7.2         8.7          5                 223         5   
4           3.7         5.2          2                 159         3   

   accidente  renunciar  ascenso_5years salario  depa_Investigación  \
0          0          1               0    bajo                   0   
1          0          1               0   medio                   0   
2          0          1               0   medio                   0   
3          0          1               0    bajo                   0   
4          0          1               0    bajo                   0   

   depa_administración  depa_contabilidad  depa_desarrollo de producto  \
0                    0                  0                            0   
1                    0                  0                            0   
2                    0                  0                            0   
3                    0                  0                            0   
4                    0                  0                            0   

   depa_informática  depa_mantenimiento  depa_mercadotecnia  \
0                 0                   0                   0   
1                 0                   0                   0   
2                 0                   0                   0   
3                 0                   0                   0   
4                 0                   0                   0   

   depa_recursos humanos  depa_soporte  
0                      0             0  
1                      0             0  
2                      0             0  
3                      0             0  
4                      0             0

In [115]:
## Convertir a dummies
dummies=pd.get_dummies(df_desercion['salario'], prefix='sal').astype(int)
dummies.head()

sal_alto  sal_bajo  sal_medio
0         0         1          0
1         0         0          1
2         0         0          1
3         0         1          0
4         0         1          0

In [116]:
dummies.describe()

sal_alto      sal_bajo     sal_medio
count  12017.000000  12017.000000  12017.000000
mean       0.082467      0.478572      0.438961
std        0.275086      0.499561      0.496281
min        0.000000      0.000000      0.000000
25%        0.000000      0.000000      0.000000
50%        0.000000      0.000000      0.000000
75%        0.000000      1.000000      1.000000
max        1.000000      1.000000      1.000000

In [117]:
#Se guardan los dummies en el dataframe y se eliminan los datos reales de salario
dummies = dummies[['sal_bajo', 'sal_medio']]
df_desercion = pd.concat([df_desercion,dummies],axis=1)
df_desercion = df_desercion.drop('salario',axis=1)

In [118]:
df_desercion.head()

satisfaccion  evaluacion  proyectos  promedio_horas_mes  semestre  \
0           3.8         5.3          2                 157         3   
1           8.0         8.6          5                 262         6   
2           1.1         8.8          7                 272         4   
3           7.2         8.7          5                 223         5   
4           3.7         5.2          2                 159         3   

   accidente  renunciar  ascenso_5years  depa_Investigación  \
0          0          1               0                   0   
1          0          1               0                   0   
2          0          1               0                   0   
3          0          1               0                   0   
4          0          1               0                   0   

   depa_administración  depa_contabilidad  depa_desarrollo de producto  \
0                    0                  0                            0   
1                    0                  0                            0   
2                    0                  0                            0   
3                    0                  0                            0   
4                    0                  0                            0   

   depa_informática  depa_mantenimiento  depa_mercadotecnia  \
0                 0                   0                   0   
1                 0                   0                   0   
2                 0                   0                   0   
3                 0                   0                   0   
4                 0                   0                   0   

   depa_recursos humanos  depa_soporte  sal_bajo  sal_medio  
0                      0             0         1          0  
1                      0             0         0          1  
2                      0             0         0          1  
3                      0             0         1          0  
4                      0             0         1          0

In [119]:
## Estadisticas Descriptivas
df_desercion.describe()

satisfaccion    evaluacion     proyectos  promedio_horas_mes  \
count  12017.000000  12017.000000  12017.000000        12017.000000   
mean       6.291537      7.168411      3.804277          200.543813   
std        2.414308      1.684195      1.165544           48.778816   
min        0.900000      3.600000      2.000000           96.000000   
25%        4.800000      5.700000      3.000000          157.000000   
50%        6.600000      7.200000      4.000000          200.000000   
75%        8.200000      8.600000      5.000000          243.000000   
max       10.000000     10.000000      7.000000          310.000000   

           semestre     accidente     renunciar  ascenso_5years  \
count  12017.000000  12017.000000  12017.000000    12017.000000   
mean       3.366231      0.154032      0.167846        0.016893   
std        1.329691      0.360994      0.373745        0.128875   
min        2.000000      0.000000      0.000000        0.000000   
25%        3.000000      0.000000      0.000000        0.000000   
50%        3.000000      0.000000      0.000000        0.000000   
75%        4.000000      0.000000      0.000000        0.000000   
max       10.000000      1.000000      1.000000        1.000000   

       depa_Investigación  depa_administración  depa_contabilidad  \
count        12017.000000         12017.000000       12017.000000   
mean             0.057752             0.036282           0.051677   
std              0.233283             0.186999           0.221383   
min              0.000000             0.000000           0.000000   
25%              0.000000             0.000000           0.000000   
50%              0.000000             0.000000           0.000000   
75%              0.000000             0.000000           0.000000   
max              1.000000             1.000000           1.000000   

       depa_desarrollo de producto  depa_informática  depa_mantenimiento  \
count                 12017.000000      12017.000000        12017.000000   
mean                      0.057086          0.081218            0.186735   
std                       0.232016          0.273181            0.389715   
min                       0.000000          0.000000            0.000000   
25%                       0.000000          0.000000            0.000000   
50%                       0.000000          0.000000            0.000000   
75%                       0.000000          0.000000            0.000000   
max                       1.000000          1.000000            1.000000   

       depa_mercadotecnia  depa_recursos humanos  depa_soporte      sal_bajo  \
count        12017.000000           12017.000000  12017.000000  12017.000000   
mean             0.056004               0.050012      0.151535      0.478572   
std              0.229939               0.217980      0.358585      0.499561   
min              0.000000               0.000000      0.000000      0.000000   
25%              0.000000               0.000000      0.000000      0.000000   
50%              0.000000               0.000000      0.000000      0.000000   
75%              0.000000               0.000000      0.000000      1.000000   
max              1.000000               1.000000      1.000000      1.000000   

          sal_medio  
count  12017.000000  
mean       0.438961  
std        0.496281  
min        0.000000  
25%        0.000000  
50%        0.000000  
75%        1.000000  
max        1.000000

### ESTIMAR MODELO COX

In [120]:
cph = CoxPHFitter().fit(df_desercion, 'semestre','renunciar')
cph.print_summary()

<lifelines.CoxPHFitter: fitted with 12017 total observations, 10000 right-censored observations>
             duration col = 'semestre'
                event col = 'renunciar'
      baseline estimation = breslow
   number of observations = 12017
number of events observed = 2017
   partial log-likelihood = -16097.40
         time fit was run = 2023-09-26 03:40:37 UTC

---
                              coef  exp(coef)   se(coef)   coef lower 95%   coef upper 95%  exp(coef) lower 95%  exp(coef) upper 95%
covariate                                                                                                                           
satisfaccion                 -0.24       0.78       0.01            -0.26            -0.22                 0.77                 0.80
evaluacion                   -0.01       0.99       0.01            -0.04             0.02                 0.96                 1.02
proyectos                    -0.29       0.75       0.02            -0.33            -0.25                 0.72                 0.78
promedio_horas_mes            0.00       1.00       0.00             0.00             0.00                 1.00                 1.00
accidente                    -1.25       0.29       0.10            -1.44            -1.05                 0.24                 0.35
ascenso_5years               -1.39       0.25       0.36            -2.09            -0.69                 0.12                 0.50
depa_Investigación           -0.39       0.68       0.12            -0.61            -0.16                 0.54                 0.85
depa_administración          -0.17       0.85       0.15            -0.45             0.12                 0.64                 1.13
depa_contabilidad            -0.14       0.87       0.10            -0.35             0.06                 0.70                 1.06
depa_desarrollo de producto  -0.03       0.97       0.10            -0.23             0.18                 0.79                 1.20
depa_informática             -0.09       0.91       0.09            -0.27             0.08                 0.76                 1.09
depa_mantenimiento            0.05       1.05       0.07            -0.08             0.18                 0.92                 1.19
depa_mercadotecnia           -0.05       0.95       0.10            -0.25             0.16                 0.78                 1.17
depa_recursos humanos         0.05       1.05       0.10            -0.15             0.25                 0.86                 1.29
depa_soporte                  0.07       1.07       0.07            -0.07             0.20                 0.93                 1.23
sal_bajo                      1.52       4.59       0.15             1.24             1.81                 3.44                 6.12
sal_medio                     1.14       3.13       0.15             0.85             1.43                 2.34                 4.18

                              cmp to      z      p   -log2(p)
covariate                                                    
satisfaccion                    0.00 -27.07 <0.005     533.76
evaluacion                      0.00  -0.69   0.49       1.03
proyectos                       0.00 -14.80 <0.005     162.17
promedio_horas_mes              0.00   3.98 <0.005      13.85
accidente                       0.00 -12.46 <0.005     115.96
ascenso_5years                  0.00  -3.90 <0.005      13.34
depa_Investigación              0.00  -3.32 <0.005      10.10
depa_administración             0.00  -1.14   0.25       1.98
depa_contabilidad               0.00  -1.39   0.17       2.59
depa_desarrollo de producto     0.00  -0.24   0.81       0.31
depa_informática                0.00  -1.04   0.30       1.74
depa_mantenimiento              0.00   0.71   0.48       1.06
depa_mercadotecnia              0.00  -0.46   0.65       0.63
depa_recursos humanos           0.00   0.48   0.63       0.66
depa_soporte                    0.00   0.94   0.35       1.52
sal_bajo                        0.00  10.37 

In [121]:
cph.hazard_ratios_

covariate
satisfaccion                   0.784878
evaluacion                     0.990475
proyectos                      0.747797
promedio_horas_mes             1.001878
accidente                      0.287708
ascenso_5years                 0.249652
depa_Investigación             0.680026
depa_administración            0.846008
depa_contabilidad              0.865134
depa_desarrollo de producto    0.974861
depa_informática               0.911053
depa_mantenimiento             1.047628
depa_mercadotecnia             0.953917
depa_recursos humanos          1.050445
depa_soporte                   1.068192
sal_bajo                       4.591047
sal_medio                      3.125465
Name: exp(coef), dtype: float64

**satisfaccion:** Su pvalue es menor que alfa, rechazamos Ho significa que si el empleado está satisfecho con su trabajo disminuye el riesgo de renuncia, por cada escala de satisfacción el riesgo de renuncia disminuye en un 78%

**evaluación:** Su pvalue es mayor que alfa, no rechazamos Ho significa que si el empleado realiza bien su trabajo no tiene efecto en el riesgo de renuncia. 

**proyectos:** Su pvalue es menor que alfa, rechazamos Ho significa que el número de proyectos asignados al empleado disminuye el riesgo de renuncia, por cada proyecto asignado disminuye el riesgo de renuncia en un 75%. 

**promedio_horas_mes:** Su pvalue es menor que alfa, rechazamos Ho significa que el promedio de horas mensuales trabajadas por el empleado aumenta el riesgo de renuncia en un 0.18%. 

**accidente:** Su pvalue es menor que alfa, rechazamos Ho significa que si el empleado ha tenido un accidente dentro de la empresa disminuye el riesgo de renuncia en un 29% comparado a si no ha tenido un accidente. 

**ascenso_5years:** Su pvalue es menor que alfa, rechazamos Ho significa que si el empleado ha sido ascendido durante los últimos 5 años disminuye el riesgo de renuncia en un 25% comparado a si no ha sido ascendido durante los últimos 5 años. 

**depa_Investigación:** Su pvalue es menor que alfa, rechazamos Ho significa que si el empleado trabaja en el departamento de investigación disminuye el riesgo de renuncia en un 68% comparado a trabajar en el departamento de ventas. 

**depa_administración, depa_contabilidad, depa_desarrollo de producto, depa_informática, depa_mantenimiento, depa_mercadotecnia, depa_recursos humanos, depa_soporte:** Su pvalue es mayor que alfa, no rechazamos Ho significa que si el empleado trabaja en cualquiera de estos departamentos no tiene efecto sobre el riesgo de renuncia. 

**sal_bajo:** Su pvalue es menor que alfa, rechazamos Ho significa que si el salario del empleado es bajo aumenta el riesgo de renuncia en un 359% comparado a si su salario es alto. 

**sal_medio:** Su pvalue es menor que alfa, rechazamos Ho significa que si el salario del empleado es medio aumenta el riesgo de renuncia en un 213% comparado a si su salario es alto. 

#### **RECOMENDACIÓN A LA EMPRESA:** 
Que traten de mantener a sus empleados de los distintos departamentos ocupados asignandoles diversos proyectos para que así ellos se sientan útiles en el trabajo y al igual se sientan satisfechos con ello, para que a su vez puedan recibir un buen salario y lograr ser ascendidos en un corto periodo. 

### SUPUESTO DE RIESGOS PROPORCIONALES

In [123]:
cph.check_assumptions(df_desercion, p_value_threshold=0.05)
# H0 viola supuesto
# Ha no viola el supuesto

The ``p_value_threshold`` is set at 0.05. Even under the null hypothesis of no violations, some
covariates will be below the threshold by chance. This is compounded when there are many covariates.
Similarly, when there are lots of observations, even minor deviances from the proportional hazard
assumption will be flagged.

With that in mind, it's best to use a combination of statistical tests and visual tests to determine
the most serious violations. Produce visual plots using ``check_assumptions(..., show_plots=True)``
and looking for non-constant lines. See link [A] below for a full example.



<lifelines.StatisticalResult: proportional_hazard_test>
 null_distribution = chi squared
degrees_of_freedom = 1
             model = <lifelines.CoxPHFitter: fitted with 12017 total observations, 10000 right-censored observations>
         test_name = proportional_hazard_test

---
                                  test_statistic      p  -log2(p)
accidente                   km              0.10   0.75      0.41
                            rank            1.43   0.23      2.10
ascenso_5years              km              3.69   0.05      4.19
                            rank            6.67   0.01      6.67
depa_Investigación          km              1.06   0.30      1.72
                            rank            5.06   0.02      5.35
depa_administración         km              3.60   0.06      4.11
                            rank            1.17   0.28      1.84
depa_contabilidad           km              0.54   0.46      1.11
                            rank            1.08   0.30      1.74
depa_desarrollo de producto km              0.07   0.80      0.33
                            rank            1.52   0.22      2.20
depa_informática            km              0.09   0.76      0.39
                            rank            3.34   0.07      3.88
depa_mantenimiento          km              0.34   0.56      0.83
                            rank            1.59   0.21      2.27
depa_mercadotecnia          km              1.23   0.27      1.90
                            rank            0.90   0.34      1.54
depa_recursos humanos       km              0.00   1.00      0.00
                            rank            0.03   0.85      0.23
depa_soporte                km              2.73   0.10      3.35
                            rank            4.33   0.04      4.74
evaluacion                  km            447.31 <0.005    327.39
                            rank          428.22 <0.005    313.60
promedio_horas_mes          km            272.76 <0.005    201.13
                            rank          307.61 <0.005    226.35
proyectos                   km            582.52 <0.005    425.12
                            rank          694.01 <0.005    505.67
sal_bajo                    km              0.51   0.48      1.07
                            rank            0.06   0.81      0.30
sal_medio                   km              0.24   0.62      0.68
                            rank            0.23   0.64      0.65
satisfaccion                km           1264.08 <0.005    917.32
                            rank          852.72 <0.005    620.30



1. Variable 'satisfaccion' failed the non-proportional test: p-value is <5e-05.

   Advice 1: the functional form of the variable 'satisfaccion' might be incorrect. That is, there
may be non-linear terms missing. The proportional hazard test used is very sensitive to incorrect
functional forms. See documentation in link [D] below on how to specify a functional form.

   Advice 2: try binning the variable 'satisfaccion' using pd.cut, and then specify it in
`strata=['satisfaccion', ...]` in the call in `.fit`. See documentation in link [B] below.

   Advice 3: try adding an interaction term with your time variable. See documentation in link [C]
below.


2. Variable 'evaluacion' failed the non-proportional test: p-value is <5e-05.

   Advice 1: the functional form of the variable 'evaluacion' might be incorrect. That is, there may
be non-linear terms missing. The proportional hazard test used is very sensitive to incorrect
functional forms. See documentation in link [D] below on how to 

[]

In [124]:
## Verificar los supuestos
cph = CoxPHFitter().fit(df_desercion, 'semestre', 'renunciar')
cph.check_assumptions(df_desercion, p_value_threshold=0.05)

The ``p_value_threshold`` is set at 0.05. Even under the null hypothesis of no violations, some
covariates will be below the threshold by chance. This is compounded when there are many covariates.
Similarly, when there are lots of observations, even minor deviances from the proportional hazard
assumption will be flagged.

With that in mind, it's best to use a combination of statistical tests and visual tests to determine
the most serious violations. Produce visual plots using ``check_assumptions(..., show_plots=True)``
and looking for non-constant lines. See link [A] below for a full example.



<lifelines.StatisticalResult: proportional_hazard_test>
 null_distribution = chi squared
degrees_of_freedom = 1
             model = <lifelines.CoxPHFitter: fitted with 12017 total observations, 10000 right-censored observations>
         test_name = proportional_hazard_test

---
                                  test_statistic      p  -log2(p)
accidente                   km              0.10   0.75      0.41
                            rank            1.43   0.23      2.10
ascenso_5years              km              3.69   0.05      4.19
                            rank            6.67   0.01      6.67
depa_Investigación          km              1.06   0.30      1.72
                            rank            5.06   0.02      5.35
depa_administración         km              3.60   0.06      4.11
                            rank            1.17   0.28      1.84
depa_contabilidad           km              0.54   0.46      1.11
                            rank            1.08   0.30      1.74
depa_desarrollo de producto km              0.07   0.80      0.33
                            rank            1.52   0.22      2.20
depa_informática            km              0.09   0.76      0.39
                            rank            3.34   0.07      3.88
depa_mantenimiento          km              0.34   0.56      0.83
                            rank            1.59   0.21      2.27
depa_mercadotecnia          km              1.23   0.27      1.90
                            rank            0.90   0.34      1.54
depa_recursos humanos       km              0.00   1.00      0.00
                            rank            0.03   0.85      0.23
depa_soporte                km              2.73   0.10      3.35
                            rank            4.33   0.04      4.74
evaluacion                  km            447.31 <0.005    327.39
                            rank          428.22 <0.005    313.60
promedio_horas_mes          km            272.76 <0.005    201.13
                            rank          307.61 <0.005    226.35
proyectos                   km            582.52 <0.005    425.12
                            rank          694.01 <0.005    505.67
sal_bajo                    km              0.51   0.48      1.07
                            rank            0.06   0.81      0.30
sal_medio                   km              0.24   0.62      0.68
                            rank            0.23   0.64      0.65
satisfaccion                km           1264.08 <0.005    917.32
                            rank          852.72 <0.005    620.30



1. Variable 'satisfaccion' failed the non-proportional test: p-value is <5e-05.

   Advice 1: the functional form of the variable 'satisfaccion' might be incorrect. That is, there
may be non-linear terms missing. The proportional hazard test used is very sensitive to incorrect
functional forms. See documentation in link [D] below on how to specify a functional form.

   Advice 2: try binning the variable 'satisfaccion' using pd.cut, and then specify it in
`strata=['satisfaccion', ...]` in the call in `.fit`. See documentation in link [B] below.

   Advice 3: try adding an interaction term with your time variable. See documentation in link [C]
below.


2. Variable 'evaluacion' failed the non-proportional test: p-value is <5e-05.

   Advice 1: the functional form of the variable 'evaluacion' might be incorrect. That is, there may
be non-linear terms missing. The proportional hazard test used is very sensitive to incorrect
functional forms. See documentation in link [D] below on how to 

[]

satisfacción y evaluación violan el supuesto de riesgo proporcional constante

In [ ]:
## Estimar el modelo AFT
## Primero tenemos que determinar si es exponencial, weibull o lognormal

In [125]:
from lifelines import ExponentialFitter, WeibullFitter, LogNormalFitter

In [126]:
mexpo = ExponentialFitter().fit(df_desercion['semestre'], df_desercion['renunciar'])
mweibull = WeibullFitter().fit(df_desercion['semestre'], df_desercion['renunciar'])
mlogn = LogNormalFitter().fit(df_desercion['semestre'], df_desercion['renunciar'])

In [127]:
mexpo.AIC_.round(2), mweibull.AIC_.round(2), mlogn.AIC_.round(2)

(16131.97, 12628.39, 11690.53)

In [ ]:
## Se elige el modelo con el menor valor
## Vamos a estimar un AFT LogNormal 

### ESTIMACIÓN AFT LOGNORMAL

In [128]:
from lifelines import LogNormalAFTFitter

In [129]:
lognAFT1 = LogNormalAFTFitter().fit(df_desercion, 'semestre', 'renunciar')
lognAFT1.print_summary()

<lifelines.LogNormalAFTFitter: fitted with 12017 total observations, 10000 right-censored observations>
             duration col = 'semestre'
                event col = 'renunciar'
   number of observations = 12017
number of events observed = 2017
           log-likelihood = -4920.18
         time fit was run = 2023-09-26 04:14:56 UTC

---
                                     coef  exp(coef)   se(coef)   coef lower 95%   coef upper 95%  exp(coef) lower 95%  exp(coef) upper 95%
param  covariate                                                                                                                           
mu_    accidente                     0.26       1.30       0.02             0.22             0.30                 1.25                 1.35
       ascenso_5years                0.37       1.44       0.06             0.24             0.49                 1.28                 1.63
       depa_Investigación            0.06       1.07       0.03             0.01             0.12                 1.01                 1.12
       depa_administración           0.11       1.12       0.03             0.05             0.18                 1.05                 1.19
       depa_contabilidad             0.02       1.02       0.03            -0.03             0.07                 0.97                 1.07
       depa_desarrollo de producto   0.01       1.01       0.02            -0.04             0.06                 0.96                 1.06
       depa_informática              0.02       1.02       0.02            -0.02             0.06                 0.98                 1.06
       depa_mantenimiento           -0.01       0.99       0.02            -0.04             0.02                 0.96                 1.02
       depa_mercadotecnia            0.03       1.03       0.03            -0.02             0.08                 0.98                 1.08
       depa_recursos humanos        -0.02       0.98       0.03            -0.07             0.03                 0.93                 1.03
       depa_soporte                 -0.02       0.98       0.02            -0.05             0.01                 0.95                 1.02
       evaluacion                    0.01       1.01       0.00            -0.00             0.01                 1.00                 1.01
       promedio_horas_mes           -0.00       1.00       0.00            -0.00            -0.00                 1.00                 1.00
       proyectos                     0.10       1.10       0.01             0.09             0.11                 1.09                 1.11
       sal_bajo                     -0.37       0.69       0.03            -0.42            -0.31                 0.66                 0.74
       sal_medio                    -0.28       0.75       0.03            -0.34            -0.23                 0.71                 0.80
       satisfaccion                  0.07       1.07       0.00             0.06             0.07                 1.06                 1.07
       Intercept                     1.28       3.58       0.04             1.20             1.35                 3.32                 3.86
sigma_ Intercept                    -1.12       0.33       0.02            -1.15            -1.09                 0.32                 0.34

                                     cmp to      z      p   -log2(p)
param  covariate                                                    
mu_    accidente                       0.00  13.04 <0.005     126.64
       ascenso_5years                  0.00   5.85 <0.005      27.61
       depa_Investigación              0.00   2.40   0.02       5.95
       depa_administración             0.00   3.47 <0.005      10.93
       depa_contabilidad               0.00   0.84   0.40       1.32
       depa_desarrollo de producto     0.00   0.51   0.61       0.71
       depa_informática                0.00   0.84   0.40       1.31
       depa_mantenimiento              0.00  -0.83   0.41       1.29
       depa_mercadotecnia

**satisfaccion:** Su pvalue es menor que alfa, rechazamos Ho significa que si el empleado está satisfecho con su trabajo disminuye el riesgo de renuncia, por cada escala de satisfacción el riesgo de renuncia disminuye en un 7%

**evaluación:** Su pvalue es mayor que alfa, no rechazamos Ho significa que si el empleado realiza bien su trabajo no tiene efecto en el riesgo de renuncia. 

**proyectos:** Su pvalue es menor que alfa, rechazamos Ho significa que el número de proyectos asignados al empleado aumenta el riesgo de renuncia, por cada proyecto asignado aumenta el riesgo de renuncia en un 10%. 

**promedio_horas_mes:** Su pvalue es menor que alfa, rechazamos Ho significa que el promedio de horas mensuales trabajadas por el empleado disminuye el riesgo de renuncia en un 0.18%. 

**accidente:** Su pvalue es menor que alfa, rechazamos Ho significa que si el empleado ha tenido un accidente dentro de la empresa aumenta el riesgo de renuncia en un 30% comparado a si no ha tenido un accidente. 

**ascenso_5years:** Su pvalue es menor que alfa, rechazamos Ho significa que si el empleado ha sido ascendido durante los últimos 5 años disminuye el riesgo de renuncia en un 44% comparado a si no ha sido ascendido durante los últimos 5 años. 

**depa_Investigación:** Su pvalue es menor que alfa, rechazamos Ho significa que si el empleado trabaja en el departamento de investigación disminuye el riesgo de renuncia en un 68% comparado a trabajar en el departamento de ventas. 

**depa_administración:** Su pvalue es menor que alfa, rechazamos Ho significa que si el empleado trabaja en el departamento de administración aumenta el riesgo de renuncia en un 12% comparado a trabajar en el departamento de ventas. 

**depa_contabilidad, depa_desarrollo de producto, depa_informática, depa_mantenimiento, depa_mercadotecnia, depa_recursos humanos, depa_soporte:** Su pvalue es mayor que alfa, no rechazamos Ho significa que si el empleado trabaja en cualquiera de estos departamentos no tiene efecto sobre el riesgo de renuncia. 

**sal_bajo:** Su pvalue es menor que alfa, rechazamos Ho significa que si el salario del empleado es bajo aumenta el riesgo de renuncia en un 69% comparado a si su salario es alto. 

**sal_medio:** Su pvalue es menor que alfa, rechazamos Ho significa que si el salario del empleado es medio aumenta el riesgo de renuncia en un 75% comparado a si su salario es alto. 

